neuralhydrology/evaluation/metrics.py

```diff
@@ -625,6 +625,45 @@ def mean_peak_timing(obs: DataArray,
     return np.mean(timing_errors) if len(timing_errors) > 0 else np.nan
 
 
+def peak_flow_bias(obs: DataArray,
+                   sim: DataArray) -> float:
+    """Calculate the percent bias of peak flows.
+
+    Uses scipy.find_peaks to find peaks in the observed time series. The observed peaks indices are used to subset
+    observed and simulated flows. Finally, the peak flow bias is calculated as the percent bias of observed peak flows
+    and corresponding simulated flows.
+
+    Parameters
+    ----------
+    obs : DataArray
+        Observed time series.
+    sim : DataArray
+        Simulated time series.
+
+    Returns
+    -------
+    float
+        Peak flow bias in percent.
+    """
+    # verify inputs
+    _validate_inputs(obs, sim)
+
+    # get time series with only valid observations
+    obs, sim = _mask_valid(obs, sim)
+
+    # heuristic to get indices of peaks and their corresponding height.
+    peaks, _ = signal.find_peaks(obs.values, distance=100, prominence=np.std(obs.values))
```

Comment:
Is this fail-safe in case L 652 returns empty arrays (e.g. all obs/sim are NaN). I know we do the same above in the peak timing function (which we should probably fix) but we don't need to do the same mistake here ;) So maybe add a check that there are at least some values and otherwise return `np.nan`?

Reply:
Actually, I tested the implementation with some saboteurs, which cause different failures. In your described case, the routine shows a runtime warning but will not fail with an error. It also ends up with an `np.nan` as a result. However, I agree that a fail-fast would be ok here, since it improves readability of the implementation by explicitly returning `np.nan` in failing cases. I will address it with a dedicated commit.

Reply:
I'm in favor of an explicit empty array edge case, given that I think the `find_peaks` function does not explicitly define its behavior for this case (in their docs).

Reply:
In the meanwhile, I've already created another commit that addressed this issue. I think, that's the reason why ;-)